### ISSUE TITLE
Bugs of reloading model

### ISSUE THREAD

**Issue by hangtingchen** [mpariente/asteroid#165]
## 🐛 Bug

Bugs of reloading model when best_k_models.json not exist

### To Reproduce
https://github.com/mpariente/asteroid/blob/0bdec2644f2d770d037ce804b7f70cb98bd5c9fa/egs/wsj0-mix/DeepClustering/model.py#L156-L157

The lines use sort to get the last model, which will performs in a incorrect way  in the following situation
```
>>> all_ckpt=['ckpt_epoch_99.ckpt','ckpt_epoch_100.ckpt','ckpt_epoch_101.ckpt']
>>> all_ckpt.sort()
>>> all_ckpt[-1]
'ckpt_epoch_99.ckpt'
```

### Expected behavior

Maybe we can use the following methods:
```
>>> all_ckpt=[(ckpt,int("".join(filter(str.isdigit,os.path.basename(ckpt))))) for ckpt in all_ckpt if ckpt.find('ckpt')>=0]
>>> all_ckpt.sort(key=lambda x:x[1])
>>> all_ckpt[-1][0]
'ckpt_epoch_101.ckpt'
```

### Environment

 - Python 3.7

**Comment by mpariente** [comment]
You're completely right, mind submitting a PR please?

**Comment by mpariente** [comment]
Solved by #166
